### ISSUE TITLE
Make sets show limited number of cards

### ISSUE THREAD

**Issue by jacob-macleod** [jacob-macleod/Flashcard-App#34]
Sets should show limited number of cards to revise and learn to avoid overwhelming the user. This should be automatically calculated from the words in each flashcard to learn today, or set my the user from "default" to another value

**Comment by jacob-macleod** [comment]
If you'd like to work on this, let me know and I'll send more info

**Comment by suyashXD** [comment]
Hi @jacob-macleod,
Could you please elaborate on the issue?
I'd like to offer my help in working on this task.

**Comment by jacob-macleod** [comment]
Sorry for the late reply! This is actually done (which I didn't realise), but in checking whether it was done, I noticed further changes to make.

See the below image:

![Image](https://github.com/user-attachments/assets/01a5ee3a-2f4d-461c-b123-c1c489c8f289)

Each "Test" flashcard has more than 19 cards (49 in total). None of the cards have been started yet, and only 19 cards are shown, as is expected. If all the cards which haven't been started yet were shown (49 cards), it would be too overwhelming for the user. This is handled on the backend, and there's a file which defines the review limits

As you can see in the image though, even though the cards in "Test" should be shown as "Not started" and shown as dark blue, they are instead tagged as "Recapping" and shown in light blue. When you click onto either "Test" set, the cards are correctly tagged as the dark blue colour:

![Image](https://github.com/user-attachments/assets/c09cb749-87ec-46ac-946e-ac942c25fa0e)

I've renamed this issue to refer to the above problem. Do you want to work on the above issue? (If not, no worries)

Thanks!

**Comment by suyashXD** [comment]
Hello, @jacob-macleod

Thanks for the detailed context. I’d love to work on this and wanted to first confirm my understanding to make sure I’m approaching it correctly.

Based on what I've learnt:

- The backend correctly limits the number of cards shown from a set (e.g., only 19 out of 49 cards that have not been started yet are shown to avoid overwhelming the user).

- However, on the Flashcards Page, these cards (not started ones) are incorrectly tagged as “Recapping” (light blue) instead of “Not Started” (dark blue).

- When you click into the actual set, the cards are correctly tagged, so the backend seems to be passing the correct data — it’s just not reflected properly in the overview view.

My assumption is that somewhere in the front-end logic (possibly the component rendering the flashcards list), the learning status is either being misinterpreted or overridden, especially when only a limited subset of the full card set is being shown.

Would it be safe to say that the issue lies in how progress state is being inferred or mapped on the overview page, rather than the data itself?

If that sounds accurate, I’ll start looking into how resolve the issue.

**Comment by jacob-macleod** [comment]
Yeah that’s correct - so I’m pretty sure it’s a front end thing based on how the progress state is being mapped onto the overview page

Thanks!

**Comment by jacob-macleod** [comment]
Thanks for your work on this @suyashXD!
